carmm/build/neb/bond_length_scan.py

```diff
@@ -43,32 +43,37 @@ def dissociation(atoms, i1, i2, step_size=0.05, n_steps=20, final_distance=None,
     from ase.constraints import FixBondLength
     import copy
     import numpy as np
+    from ase.neighborlist import natural_cutoffs
+
+    # retrieve natural cutoffs for atomic radii
+    cutOff = natural_cutoffs(atoms, mult=2)
 
     # retrieve initial atom - atom distance and atom[i2] position
     pos_diff = atoms[i1].position - atoms[i2].position
     initial_dist = np.linalg.norm(pos_diff)
     initial_i2_pos = copy.deepcopy(atoms[i2].position)
-    z_diff = 2.0
+
     # retrieve z-coordinate for z_bias
     if z_bias:
+        # Define a list of atom chemical symbols and their count
+        # Take the z coordinate of the most abundant element in the surface slab
+        chem_symbol_count = [[x, atoms.get_chemical_symbols().count(x)] for x in set(atoms.get_chemical_symbols())]
+        # Sort the list by the count
+        chem_symbol_count.sort(key=lambda k: k[1])
+        surf_z_list = [atom.z for atom in atoms if atom.symbol == chem_symbol_count[-1][0] and atom.tag > 0]
```

Comment:
I suspect atoms.tag makes sense within a greater workflow, but I would probably include some assertion error early on, prodding the user to define it if they haven't done so already.

Reply:
I removed the use of tags and replaced them with avoiding z-coordinates of atoms being moved. The only edge case I can think of is e.g. metal oxides with more oxygen than metal and large adsorbates containing oxygen, where not all oxygens are being moved. Then potentially the dynamically applied `z_bias` might be incorrect, but can always be replaced by a float.